mail/api.py

```diff
@@ -114,9 +126,14 @@ def send_batch(cls, subject, body, recipients,  # pylint: disable=too-many-argum
             chunk_size (int): The maximum amount of emails to be sent at the same time
 
         Returns:
-            list: List of requests.Response HTTP response from Mailgun
+            list:
+                List of (recipients, requests.Response, exception) where
+                recipients is a list of recipient emails,
+                Response is an HTTP response from Mailgun, if there is one,
+                exception is an exception which occurred when attempting to mail, if there is one
```

Comment:
Why change the return values here? I think it's fine to have this function continue to return a list of HTTP responses, since any caller that wants to know if there were exceptions can simply iterate over those responses to see if there were any failures. It's no different from checking the return values for exception objects.